python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -909,9 +981,12 @@ async def _check_errors_and_send_events_shutter(
             fault_code = ", ".join(
                 [f"{message['code']}={message['description']}" for message in messages]
             )
-            await self.evt_shutterEnabled.set_write(
-                state=EnabledState.FAULT, faultCode=fault_code
-            )
+            # Check supported event to make sure of backward compatibility with
+            # XML 12.0.
+            if support_dds_item("evt_shutterEnabled"):
```

Comment:
Again, use `hasattr` to avoid parsing the idl file.

Reply:
Done